spec/requests/list_controller_spec.rb

```diff
@@ -1283,6 +1283,35 @@
       end
     end
 
+    it "should filter with tag_group option" do
+      topic_with_tag = Fabricate(:topic, tags: [tag])
+      topic2_with_tag = Fabricate(:topic, tags: [tag])
+      Fabricate(:tag_group, tags: [tag], name: "tag_group")
+
+      sign_in(user)
+
+      get "/filter.json", params: { q: "tags:tag1" }
+
+      expect(response.status).to eq(200)
+
+      parsed = response.parsed_body
+
+      expect(parsed["topic_list"]["topics"].length).to eq(2)
+      expect(parsed["topic_list"]["topics"].map { |topic| topic["id"] }).to contain_exactly(
+        topic_with_tag.id,
+        topic2_with_tag.id,
+      )
```

Comment:
I'm not sure if we need these assertions since it is already covered by another test.